flows/engine/session.go

```diff
@@ -7,24 +7,36 @@ import (
 	"github.com/nyaruka/goflow/flows"
 	"github.com/nyaruka/goflow/flows/events"
 	"github.com/nyaruka/goflow/flows/runs"
+	"github.com/nyaruka/goflow/flows/waits"
 	"github.com/nyaruka/goflow/utils"
 )
 
+// used to trigger a new flow run in the event loop
+type flowTrigger struct {
```

Comment:
We can chat naming here - but the basic idea is that we need a place on session to communicate back to the event loop that it should switch to a new flow (or just start the main flow)